src/modules/10-common/components/GitPopoverV2/GitPopoverV2.tsx

```diff
@@ -131,15 +135,21 @@ export const GitPopoverV2 = ({
     }
   }, [forceFetch, branchChangeDisabled, refetch])
 
+  useEffect(() => {
+    //here, default branch is set which is getting consumed in template details for setting it in selected template gitdetails object
+    if (response?.data?.defaultBranch?.name) setDefaultBranch?.(response?.data?.defaultBranch?.name)
```

Comment:
NIT: This optional chaining can be removed as the value is already checked. 
```suggestion
    if (response?.data?.defaultBranch?.name) setDefaultBranch?.(response.data.defaultBranch.name)
```